modules/shared/src/main/scala/org/tessellation/security/Hash.scala

```diff
@@ -52,19 +56,34 @@ object hash {
 }
 
 trait Hashable[F[_]] {
-  def hash[A <: AnyRef](data: A): Either[Throwable, Hash]
+  def hash[A: Encoder](data: A): F[Hash]
 }
 
 object Hashable {
```

Comment:
I am not sure if `Hashable` is a valid name anymore. Previously it was used on "hashable" things but now it is a type-class so I am thinking about a different name like: "Hasher" or sth. WDYT?

Reply:
@TheMMaciek suggested that the name looks fine as it represents some capability. So you add the capability to the effect type. I can change the name if you want, but I cannot think a better alternative tbh

Reply:
Now that I look at it I think it may make sense to have it called `Hasher`, it would make sense to have it called `Hashable` if we would have it on the object it hashes (A) rather than on `F[_]`. Calling it something like `HashingAble` would reflect the functionality more, but it sounds bad.